File: test-suite/src/test/java/io/micronaut/docs/function/aws/test/RequestHandlerTest.java
package io.micronaut.docs.function.aws.test;

//tag::clazz[]
import io.micronaut.context.ApplicationContext;
import io.micronaut.function.aws.test.annotation.MicronautLambdaTest;
import org.junit.jupiter.api.Test;

import jakarta.inject.Inject;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautLambdaTest
public class RequestHandlerTest {
    @Inject
    private ApplicationContext context;

    @Test
    void testHandler() {
        SampleRequestHandler sampleRequestHandler = new SampleRequestHandler(context);
        assertEquals("Hello world", sampleRequestHandler.execute("world"));
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/aws/test/RequestHandlerTest.kt
package io.micronaut.docs.function.aws.test

//tag::clazz[]
import io.micronaut.context.ApplicationContext
import io.micronaut.function.aws.test.annotation.MicronautLambdaTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautLambdaTest
class RequestHandlerTest {
    @Inject
    lateinit var context: ApplicationContext

    @Test
    fun testHandler() {
        val sampleRequestHandler = SampleRequestHandler(context)
        assertEquals("Hello world", sampleRequestHandler.execute("world"))
    }
}
//end::clazz[]
